test/e2e/llmisvc/test_llm_inference_service.py

```diff
@@ -0,0 +1,297 @@
+        except Exception as e:
+            raise AssertionError(f"Failed to get service URL: {e}") from e
+
+        completion_url = f"{service_url}/v1/completions"
+        test_payload = {"model": model_name, "prompt": "test", "max_tokens": 1}
+
+        try:
+            response = requests.post(
+                completion_url,
+                headers={"Content-Type": "application/json"},
+                json=test_payload,
+                timeout=30,
+            )
+        except Exception as e:
+            raise AssertionError(f"Failed to call model: {e}") from e
+
+        assert (
+            response.status_code == 200
+        ), f"Service returned {response.status_code}: {response.text}"
+        return service_url
+
+    return wait_for(assert_model_responds, timeout=timeout_seconds, interval=10.0)
+
+
+def get_llm_service_url(
+    kserve_client, service_name, namespace, version=constants.KSERVE_V1ALPHA1_VERSION
+):
+    try:
+        llm_isvc = get_llmisvc(kserve_client, service_name, namespace, version)
+
+        if "status" not in llm_isvc:
+            raise ValueError(f"No status found in LLM inference service {service_name}")
+
+        status = llm_isvc["status"]
+
+        if "url" in status and status["url"]:
+            return status["url"]
+
+        if (
+            "addresses" in status
+            and status["addresses"]
+            and len(status["addresses"]) > 0
+        ):
+            first_address = status["addresses"][0]
+            if "url" in first_address:
+                return first_address["url"]
+
+        raise ValueError(f"No URL found in LLM inference service {service_name} status")
+
+    except Exception as e:
+        raise ValueError(
+            f"Failed to get URL for LLM inference service {service_name}: {e}"
+        ) from e
+
+
+def wait_for(assertion_fn, timeout: float = 5.0, interval: float = 0.1):
+    deadline = time.time() + timeout
+    while True:
+        try:
+            return assertion_fn()
+        except AssertionError:
+            if time.time() >= deadline:
+                raise
+            time.sleep(interval)
+
+
+def get_model_name_from_configs(config_names):
+    """Extract model name from model config."""
+    for config_name in config_names:
+        if config_name.startswith("model-"):
+            config = LLMINFERENCESERVICE_CONFIGS[config_name]
+            if "model" in config and "name" in config["model"]:
+                return config["model"]["name"]
+    return "default-model"
+
+
+def generate_service_name(test_name, config_names):
+    base_name = test_name.split("[")[0]  # Remove everything after [
+    base_name = base_name.replace("test_", "").replace("_", "-")
+    config_suffix = "-".join(sorted(config_names))
+    service_name = f"{base_name}-{config_suffix}"
+    service_name = service_name.lower()
+    service_name = service_name[:63].rstrip("-")
+    return service_name
+
+
+def collect_diagnostics(service_name, namespace):
```

Comment:
This is definitely a good start, as a follow up maybe we can have a generic "collector" that collects also other "related objects" (deployments, pods and **pod logs**, inferencepools, httproutes, etc)  using the labels we're using to label objects related to llmisvc (part-of and component/name) since sometimes just the status doesn't describe the full picture (for example when a pod failure we would only get a generic error in the status propagated from Deployement)

Reply:
Done in eb081ef